planning/behavior_path_planner/src/utils/start_planner/shift_pull_out.cpp

```diff
@@ -155,17 +155,26 @@ std::vector<PullOutPath> ShiftPullOut::calcPullOutPaths(
     start_planner_utils::calcEndArcLength(s_start, forward_path_length, road_lanes, goal_pose);
   const double s_end = path_end_info.first;
   const bool path_terminal_is_goal = path_end_info.second;
-
   constexpr double RESAMPLE_INTERVAL = 1.0;
   PathWithLaneId road_lane_reference_path = utils::resamplePathWithSpline(
     route_handler.getCenterLinePath(road_lanes, s_start, s_end), RESAMPLE_INTERVAL);
+  const double shift_distance = s_end - s_start;
```

Comment:
this is arc length in road lane, better to use shifted path length

Reply:
@kosuke55 
Thank you so much for your check :heart: 
I think my bad variable name causes confusion.
Here, I want to calculate ditance between shift_start pose and shift_end pose along the non_shifted path.
And in my understanding, non_shifted_path is same with road_reference_path, correct?
Also I mistalenly used goal pose not shift_end_pose....
In this case in my opinion,
1. I should change variable name to `non_shifted_path_arc_length`.
2. I should use `shift_end_pose_ptr->position` not `goal_pose.position` as you suggested another comment.
So here is my proposal to revised code. (will clean if my understanging is correct)
```
  // non_shifted_path for when shift length or pull out distance is too short
  const PullOutPath non_shifted_path = std::invoke([&]() {
    PullOutPath non_shifted_path{};
    // The assumption for this calculation of velocity and acceleration is to accelerate with
    // constant acceleration until the road velocity limit is reached at the shift end point.
    const auto & arc_position_non_shifted_path_end =
      getArcCoordinates(road_lanes, shift_end_pose_ptr->position);
    const double non_shifted_path_arc_length = arc_position_non_shifted_path_end.length - s_start;
    const size_t non_shifted_path_end_idx =
    findNearestIndex(road_lane_reference_path.points, shift_end_pose_ptr->position);
    const double terminal_velocity =
      road_lane_reference_path.points.at(non_shifted_path_end_idx).point.longitudinal_velocity_mps;
    const double time_to_end_pose = shift_distance / terminal_velocity;
    non_shifted_path.partial_paths.push_back(road_lane_reference_path);
    non_shifted_path.start_pose = start_pose;
    non_shifted_path.end_pose = start_pose;
    non_shifted_path.acceleration = 2 * shift_distance / std::pow(time_to_end_pose, 2);
    non_shifted_path.terminal_velocity = terminal_velocity;
    return non_shifted_path;
  });
 ```

Reply:
@kyoichi-sugahara 

> And in my understanding, non_shifted_path is same with road_reference_path, correct?

yes, correct!

And sorry, I don't understand about the proposal code.
- Why shifted_end_pose_ptr is used for non_shifted_path?
- The end_pose of non_shifted_path matches the start_pose, but to what extent are you trying to safety check?
- When the ego is close to the center line, I think isExecutionRequest will be set to false in the future, but what role will non_shifted_path play? (Isn't it just a guard?)